src/filters/PaddingSides.ts

```diff
@@ -0,0 +1,164 @@
+import type { Filter } from './Filter';
+
+/**
+ * class used for paddings of text styles and filters
+ * @category filters
+ * @advanced
+ */
+export class PaddingSides
+{
+    public top: number = 0;
+    public right: number = 0;
+    public bottom: number = 0;
+    public left: number = 0;
+
+    get horizontal()
+    {
+        return this.left + this.right;
+    }
+
+    get vertical()
+    {
+        return this.top + this.bottom;
+    }
+
+    public copyFrom(sides: IPaddingSidesLike): this
+    {
+        if (!sides)
+        {
+            this.top = 0;
+            this.right = 0;
+            this.bottom = 0;
+            this.left = 0;
+        }
+        else if (typeof sides === 'number')
+        {
+            this.top = sides;
+            this.right = sides;
+            this.bottom = sides;
+            this.left = sides;
+        }
+        else if (sides instanceof Array)
+        {
+            this.top = (sides as any)[0];
+            this.right = (sides as any)[1];
+            this.bottom = (sides as any)[2];
+            this.left = (sides as any)[3];
```

Comment:
Please do not cast any here. Consider adding a type-guard function instead to keep from using any.

```ts
function isNumberList(value: unknown): value is [number, number, number, number] {
  return value instanceof Array
    && value.length === 4
    && value.every(v => typeof v === 'number');
}
```